### PR TITLE
[DOCU-2757] CP/DP forward proxy config

### PR DESCRIPTION
**Pull request by lena-larionova**
### Summary
Doc for setting up a forward proxy connection between CP and DP.

I can't really test this, so I can't verify that this is correct. Need some SME help.

Outstanding questions @dndx:
* From design doc: "Customers need to ensure their proxy does not MITM or mutate the certificate used by Kong DP when connecting to CP" - How do you verify this? 
* What special info do we need for Konnect here?
* From design doc: "We should document roughly how many connections each Kong DP will open to the CP, so that user knows how much proxy servers are needed to support a given number of Kong deployments." - how would you estimate this number?

### Reason
https://konghq.atlassian.net/browse/DOCU-2757

### Testing
https://deploy-preview-4846--kongdocs.netlify.app/gateway/latest/production/networking/cp-dp-proxy/

### CODE REVIEW

File: src/gateway/production/networking/cp-dp-proxy.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+* `proxy_server`: Proxy server defined as a URL. {{site.base_gateway}} will
+only use this option if any component is explicitly configured to use the proxy.
+
+* `proxy_server_tls_verify`: Toggles server certificate verification if
+`proxy_server` is in HTTPS. Set to `on` if using HTTPS (default), or `off` if
+using HTTP.
+
+* `cluster_use_proxy`: Tell the cluster to use HTTP CONNECT proxy support for
```
Comment (cloudjumpercat) [inline src/gateway/production/networking/cp-dp-proxy.md:35]:
```suggestion
* `cluster_use_proxy`: Tells the cluster to use HTTP CONNECT proxy support for
```

**Review by cloudjumpercat — APPROVED** [review]
Just one small suggestion, otherwise LGTM!

**Comment by lena-larionova** [comment]
Merging without SME review for now.

**Comment by rosskukulinski** [comment]
@lena-larionova are there any plans to have similar content in the Konnect documention?

**Comment by lena-larionova** [comment]
> @lena-larionova are there any plans to have similar content in the Konnect documention?

There are, but I don't currently have enough info to write the Konnect piece. One of the outstanding questions is still "What special info do we need for Konnect here?"

**Pull request merged**